api/openapi-spec/swagger.json

```diff
@@ -5033,6 +5040,9 @@
    "v1.PersistentVolumeAccessMode": {},
    "v1.PersistentVolumeClaimSpec": {
     "description": "PersistentVolumeClaimSpec describes the common attributes of storage devices and allows a Source for provider-specific attributes",
+    "required": [
+     "dataSource"
```

Comment:
Ok, this is an issue for us. I tracked down where it is coming from and field a PR against k8s https://github.com/kubernetes/kubernetes/pull/72768. They have the `+optional` field which we don't yet interpret. We rely on `omitempty` right now.

In the future we should interpret `+optional` too. Right now we can override the `required`  in https://github.com/kubevirt/kubevirt/blob/master/pkg/util/openapi/openapi.go#L62.

Reply:
Oh thanks for the hint!
I tried to override `required` in the function you mentioned by adding the following to the for loop:
```
if k == "v1.PersistentVolumeClaimSpec" {
    s.Required = make([]string, 0)
}
```
But this doesn't seem to work... Any idea how I can solve this?

Reply:
Maybe you have to set `s.SchemaProps.Required`?

Reply:
The most correct solution would probably be to iterate through the string list and explicitly filter out `dataSource`.

Reply:
Thanks @rmohr ! I got it to work now and I filter out `dataSource`.

Though I wonder why the other changes in the function do not need to set `openapispec.Definitions`. It seems to me like they should have the same problem...

Reply:
We modify copies since the maps don't container pointers to values (that's how the structs from the openapi libraray are provided) and then place them in the maps again. Since the maps are pointers, that updates the maps everywhere. A little bit weird at first but makes sense.